// mailchimp-lite/dist/v2.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

var _get = function get(_x2, _x3, _x4) { var _again = true; _function: while (_again) { var object = _x2, property = _x3, receiver = _x4; _again = false; if (object === null) object = Function.prototype; var desc = Object.getOwnPropertyDescriptor(object, property); if (desc === undefined) { var parent = Object.getPrototypeOf(object); if (parent === null) { return undefined; } else { _x2 = parent; _x3 = property; _x4 = receiver; _again = true; desc = parent = undefined; continue _function; } } else if ('value' in desc) { return desc.value; } else { var getter = desc.get; if (getter === undefined) { return undefined; } return getter.call(receiver); } } };

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _api = require('./api');

var _api2 = _interopRequireDefault(_api);

var V2 = (function (_API) {
  _inherits(V2, _API);

  function V2() {
    _classCallCheck(this, V2);

    _get(Object.getPrototypeOf(V2.prototype), 'constructor', this).apply(this, arguments);
  }

  _createClass(V2, [{
    key: 'prepareRequestOptions',
    value: function prepareRequestOptions(endpoint) {
      var data = arguments.length <= 1 || arguments[1] === undefined ? {} : arguments[1];

      data.apikey = this.key;
      return {
        uri: '' + this.apiRoot + endpoint + '.json',
        body: data,
        json: true
      };
    }

    // mailchimp only supports POST
    // > All API calls should be made with HTTP POST.
    // https://apidocs.mailchimp.com/api/2.0/#rest-vs-rpc
  }, {
    key: 'post',
    value: function post(endpoint, data) {
      return this.request('post', endpoint, data);
    }
  }]);

  return V2;
})(_api2['default']);

exports['default'] = V2;
module.exports = exports['default'];
